### PR TITLE
TSK-1866: Refactor UpdateClassificationAccTest to use Test-API

### PR DESCRIPTION
**Pull request by ensarevlideveloper**
Sonar:
* https://sonarcloud.io/summary/new_code?id=ensarevlideveloper_taskana&branch=TSK-1866
---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [X] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [ ] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [X] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [X] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [ ] Commit message format → TSK-XXX: Your commit message.
- [ ] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [ ] SonarCloud analysis meets our standards
- [ ] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [ ] Readability

### CODE REVIEW

File: lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java [hunk 1]
```diff
@@ -0,0 +1,834 @@
...
+      assertThatThrownBy(() -> classificationService.updateClassification(classification))
+          .usingRecursiveComparison()
+          .isEqualTo(expectedException);
+    }
+
+    /**
+     * The classification used here is created in the {@linkplain #createClassifications method} in
+     * the beginning.
+     *
+     * @throws Exception for errors in the updating or reading process of entities.
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java:659]: Do we need this comment?

File: lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java [hunk 2]
```diff
@@ -0,0 +1,834 @@
...
+                  "The Classification '%s' has the same key and parent key",
+                  classification.getName()));
+      assertThatThrownBy(() -> classificationService.updateClassification(classification))
+          .usingRecursiveComparison()
+          .isEqualTo(expectedException);
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @Test
+    void should_NotThrowException_When_UpdatingClassificationWithEmptyServiceLevel()
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java:757]: What do you think about moving this test to the top class or to `UpdatePriorityAndServiceLevelTest `and checking if the serviceLevel is updated to "" correctly?

File: lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java [hunk 3]
```diff
@@ -0,0 +1,834 @@
...
+  @TestInstance(Lifecycle.PER_CLASS)
+  @Nested
+  class UpdateClassificationCategoryTest {
+    Classification classification;
+    Task task;
+    Instant createdBefore;
+    Instant modifiedBefore;
+
+    /**
+     * User Role is admin here since we create and get Tasks.
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java:776]: UserRole is not admin, but businessadmin

File: lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java [hunk 4]
```diff
@@ -0,0 +1,834 @@
...
+      assertThat(expectedUpdatedTask)
+          .usingRecursiveComparison()
+          .ignoringFields("modified")
+          .isEqualTo(updatedTask);
+      assertThat(expectedUpdatedTask.getModified()).isAfterOrEqualTo(modifiedBefore);
+    }
+
+    @WithAccessId(user = "businessadmin")
+    @Test
+    void shouldUpdateTask_When_UpdatingClassificationCategory() throws Exception {
```
Comment (ryzheboka) [inline lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java:821]: There should be an underscore after `should`. Where do we check update of a Task in this test?

File: lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java (hunk 1 above)
Comment (ensarevlideveloper) [reply lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java:659]: I think it's a good to know. Especially, when a new contributor is going through the code, he/she doesn't need to search where or when it was created.

Comment (ensarevlideveloper) [reply lib/taskana-core-test/src/test/java/acceptance/classification/update/UpdateClassificationAccTest.java]: done :)

**Pull request merged**